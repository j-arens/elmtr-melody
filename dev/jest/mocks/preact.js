module.exports = {
    unmountComponent: jest.fn(),
};
