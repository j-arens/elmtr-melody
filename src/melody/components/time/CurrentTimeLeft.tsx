import { WithOptionalClassName } from '@components/type';
import { h } from 'preact';
import TimeProvider from './TimeProvider/';

interface Props extends WithOptionalClassName {
    currentTime: number;
    duration: number;
    getTime: (time: number) => string;
}

export const CurrentTimeLeft = ({ currentTime, duration, getTime, className = '' }: Props) => (
    <span class={className}>
        {getTime(duration - currentTime)}
    </span>
);

export default TimeProvider(CurrentTimeLeft);
