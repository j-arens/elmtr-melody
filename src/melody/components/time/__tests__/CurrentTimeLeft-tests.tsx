import { h } from 'preact';
import { render } from 'preact-render-to-string';
import { CurrentTimeLeft } from '../CurrentTimeLeft';

describe('CurrentTimeLeft', () => {
    it('renders', () => {
        const component = render(
            <CurrentTimeLeft
                className="yas-marina-blue"
                currentTime={0}
                duration={0}
                getTime={jest.fn().mockReturnValueOnce(80)}
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
