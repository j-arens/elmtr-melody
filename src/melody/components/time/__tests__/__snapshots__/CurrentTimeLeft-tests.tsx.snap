// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentTimeLeft renders 1`] = `<span class="yas-marina-blue" data-cy="time-left">80</span>`;
